internal/syntaxutil/README.md

```diff
@@ -0,0 +1,58 @@
+# vendored std regexp/syntax
+
+This package contains a vendored copy of std regexp/syntax. However, it only
+contains the code for converting syntax.Regexp into a String. It is the
+version of the code at a recent go commit, but with a commit which introduces
+a significant performance regression reverted.
+
+At the time of writing regexp.String on go1.22 is taking 40% of CPU at
+Sourcegraph. This should return to ~0% with this vendored code.
+
+https://github.com/sourcegraph/sourcegraph/issues/61462
+
+## Vendored commit
+
+```
+commit 2e1003e2f7e42efc5771812b9ee6ed264803796c
```

Comment:
Questions

1. Should we file this as a perf bug on the go   package itself also? 
2. What functionality do we lose rolling back that commit?

Reply:
It is worth filing an issue just for the discussion. But I suspect they will close it as wontfix since apparently String() is only meant to be used for debug. However, there is a non-debug case where you wan't to take a syntax.Regexp and get a compiled Regexp (which is our use case). So that could atleast spark discussion for a new API to use.

The functionality lost is honestly not important. The "debug" strings are nicer for a human to look at... but not that different. In practice for us no change.

Reply:
@keegancsmith  Thanks and great tactical change

Reply:
+1 to filing an issue when you have the time, always nice to engage with downstream projects and maybe it will help us avoid the need for a future fork!